### DISCUSSION TITLE
No volume?

### DISCUSSION THREAD

**Discussion by Huecuva** [jcorporation/myMPD#473] [Q&A] [answered]
I have an mpd server set up and running and I installed mympd on it. I managed to access the webgui from a different machine on my lan and am able to see and play the music I have on the mpd server. However, there is no audio and the volume controls are disabled for some reason. I have been able to play my mpd stream through other means, like using MPD Remote from my phone, Foobar2000, Clementine or even just a browser tab pointed directly at the mpd server IP address. I just can't get any sound out of mympd.

Does anyone know how to fix this? Thanks.

EDIT: I've just installed it on another MPD server I have running. The first is in a VM and I did not have an audio device passed through into the VM. However, the audio still did not work after I had configured the VM with an audio device. It is this second mpd server that is running on a bare metal Debian server that I have been able to listen to previously. However, I also have an Apache server running on this Debian server and it appears to be interfering with the webGUI for myMPD. In my browser I can simply go to the VM's IP address to see the myMPD webGUI, however going to my physical MPD server's IP address brings up the Apache server and specifying the port I configured in /etc/mympd.conf doesn't work either.

Any help is appreciated.

EDIT AGAIN: Okay so it turns out mympd just wasn't running on the physical server. I can now access the webGUI for it. However, I still cannot get audio from the webGUI. Is that not something that is supposed to happen?

**Comment by jcorporation** [comment]
You should configure MPD outputs according to your configuration, that is not the job of myMPD.

myMPD can play the MPD http stream: https://github.com/jcorporation/myMPD/wiki/Local-playback

**Reply by Huecuva** [reply]
Thanks, that was helpful. However, I still can't get it to work. I've added the `streamport = 8000` line to my /etc/mympd.conf file and commented out the `bind_to_address` line in the audio output section of /etc/mpd.conf as well as adding the `format` and `always_on` lines from that page. The rest of my audio output section was the same. However, I am still unable to get any audio out of mympd. I was also finally able to get audio out of a browser tab directed right to the stream IP address before I commented it out to try the local streaming, so I know the VM is capable of producing audio. From my understanding, this makes the mpd stream to 127.0.0.1 and mympd transmits it through the browser over whatever port I configure it to show up, right? It's not working for me and the mympd still says volumecontrol is disabled.

Here is the [mpd] section of my /etc/mympd.conf:

> [mpd]
> Connection to mpd, unix socket or host/port, socket preferred
> host = 127.0.0.1
> port = 6000
> streamport = 8000
> host = /run/mpd/socket

And here is the audio_output section of my /etc/mpd.conf:

> audio_output {
>         encoder                 "lame"
>         type                    "httpd"
>         name                    "Radio"
>         format                  "44100:16:1"
>         port                    "8000"
>         tags                    "yes"
>         always_on               "yes"
>         bitrate                 "128"
> }

I have also tried adding a bind_to_address in the audio_output section binding to 127.0.0.1. It still isn't working.

EDIT: When I run `systemctl status mympd` I get the following errors:

> Apr 21 21:52:06 radio mympd[651]: WARN     mpdworker Albumcache, skipping "Modern Rock/INXS_-_Mystify.mp3"
> Apr 21 21:52:06 radio mympd[651]: WARN     mpdworker Albumcache, skipping "Modern Rock/INXS_-_Suicide Blonde.mp3"
> Apr 21 21:52:06 radio mympd[651]: WARN     mpdworker Albumcache, skipping "Modern Rock/INXS_-_Beautiful Girl.mp3"
> Apr 21 21:52:06 radio mympd[651]: WARN     mpdworker Albumcache, skipping "White Noise/White Noise  Ten Hours - Ambient
> Apr 21 21:52:06 radio mympd[651]: 140354697250560:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unApr 21 21:52:06 radio mympd[651]: 2021-04-22 04:52:06  E mongoose.c:411:mg_error   3 tls hs: rc -1, err 6
> Apr 21 21:52:06 radio mympd[651]: 140354697250560:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unApr 21 21:52:06 radio mympd[651]: 2021-04-22 04:52:06  E mongoose.c:411:mg_error   5 tls hs: rc -1, err 6
> Apr 21 21:52:06 radio mympd[651]: 140354697250560:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unApr 21 21:52:06 radio mympd[651]: 2021-04-22 04:52:06  E mongoose.c:411:mg_error   6 tls hs: rc -1, err 6

**Reply by Huecuva** [reply]
I went through the wiki and added all the extra things I needed to my /etc/mpd.conf file. Here is the whole thing now:

> music_directory                 "/home/testradio/music/0/"
> playlist_directory              "/home/testradio/music/"
> #db_file                                "/var/lib/mpd/tag_cache"
> log_file                        "/var/lib/mpd/mpd.log"
> pid_file                        "/run/mpd/pid"
> state_file                      "/var/lib/mpd/state"
> sticker_file                    "/var/lib/mpd/sticker.sql"
> metadata_to_use                 "artist,title,album,track"
> auto_update                     "yes"
> # bind_to_address               "127.0.0.1"
> bind_to_address                 "/run/mpd/socket"
> port                            "6000"
> 
> database {
>         plugin                  "simple"
>         path                    "/var/lib/mpd/tag_cache"
>         cache_directory         "/bar/lib/mpd/cache"
> }
> 
> neighbors {
>         plugin                  "smbclient"
> }
> neighbors {
>         plugin                  "udisks"
> }
> neighbors {
>         plugin                  "upnp"
> }
> 
> audio_output {
>         encoder                 "lame"
>         type                    "httpd"
>         name                    "Radio"
>         format                  "44100:16:1"
> #       bind_to_address         "10.192.10.154"
>         bind_to_address         "127.0.0.1"
>         port                    "8000"
>         tags                    "yes"
>         always_on               "yes"
>         bitrate                 "128"
> }

I've also commented out the host 127.0.0.1 line in /etc/mympd.conf so it only has the /run/mpd/socket.

It's still not working.

**Comment by jcorporation** [comment, marked as answer]
myMPD proxies the stream from /stream uri to 127.0.01:8000 as documented. You must enable the output in the volume menu and the local playback in the settings dialog.

MPD disables volume control if no output with a mixer is present, this is no myMPD setting.

The ssl errors from mongoose are strange, please attach debug logs from startup of myMPD.

**Reply by Huecuva** [reply]
Okay, I've enabled the local playback in the settings. I didn't know there was a settings menu. That explains a lot. Thank you. However, volumecontrol is still disabled and the volume is at minimum so I am still not getting any audio. I don't know how to make mpd enable the volume control. What do you mean by an output with a mixer?

Don't worry about those errors. They don't seem to be appearing anymore.

EDIT: Nevermind. It seems to be working now. No idea why it suddenly decided to work, but it is.

Thanks!

**Reply by jcorporation** [reply]
Mixer: https://www.musicpd.org/doc/html/user.html#configuring-audio-outputs
